domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt

```diff
@@ -851,6 +855,47 @@ class ProfileManagementController @Inject constructor(
     }
   }
 
+  /**
+   * Sets the last selected [classroomId] for the specified [profileId]. Returns a [DataProvider]
+   * indicating whether the save was a success.
+   */
+  fun updateLastSelectedClassroomId(
+    profileId: ProfileId,
+    classroomId: String
+  ): DataProvider<Any?> {
+    val deferred = profileDataStore.storeDataWithCustomChannelAsync(
+      updateInMemoryCache = true
+    ) { profileDatabase ->
+      val profile = profileDatabase.profilesMap[profileId.internalId]
+      val updatedProfile = profile?.toBuilder()?.setLastSelectedClassroomId(
+        classroomId
+      )?.build()
+      val profileDatabaseBuilder = profileDatabase.toBuilder().putProfiles(
+        profileId.internalId,
+        updatedProfile
+      )
+      Pair(profileDatabaseBuilder.build(), ProfileActionStatus.SUCCESS)
+    }
+    return dataProviders.createInMemoryDataProviderAsync(
+      SET_LAST_SELECTED_CLASSROOM_ID_PROVIDER_ID
+    ) {
+      return@createInMemoryDataProviderAsync getDeferredResult(profileId, null, deferred)
+    }
+  }
+
+  /**
+   * Returns a [DataProvider] containing a nullable last selected classroom ID for the specified
+   * [profileId].
+   */
```

Comment:
So, this was done following the official kotlin guidelines here: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments.

It says to avoid using `@params` to define the kdocs for the parameters passed. In this case `[profileId]` is passed as a parameter to the function.